### PR TITLE
Added libvirt docs for firecracker driver

### PR DESCRIPTION
**Pull request by felixp72**
Signed-off-by: Felix Puscasu <fpuscasu@amazon.com>

### CODE REVIEW

File: docs/meson.build [hunk 1]
```diff
@@ -112,6 +112,7 @@ docs_rst_files = [
   'developer-tooling',
   'drvqemu',
   'drvch',
+  'drvfc',
```
Comment (alindima) [inline docs/meson.build:115]: is the html built from the rst file?
Shouldn't we commit the html file as well?

File: docs/drvfc.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
+=======================
+Firecracker driver
+=======================
+
+.. contents::
+
+Firecracker is an open source Virtual Machine Monitor (VMM) that
```
Comment (alindima) [inline docs/drvfc.rst]: you may add a link here to the repo

File: docs/drvfc.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+``fc:///system``
+
+
+Example guest domain XML configurations
+=======================================
+
+The Firecracker driver in libvirt is in its early stage of development
+supporting a subset of the features provided by the Firecracker VMM, for now being
+more a proof of concept
```
Comment (alindima) [inline docs/drvfc.rst]: overall: add periods at the end of paragraphs

File: docs/drvfc.rst [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+Firecrackeer binaries can be found on the
```
Comment (alindima) [inline docs/drvfc.rst]:
```suggestion
Firecracker binaries can be found on the
```

File: docs/drvfc.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+`github repository <https://github.com/firecracker-microvm/firecracker>`__
```
Comment (alindima) [inline docs/drvfc.rst]: you can link the releases page

File: docs/drvfc.rst [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+The `<root>` tag specifies which of the following disk devices is used as the rootfs by matching the
+dev mount point attribute in the `<target>` tag.
+
+For the devices section:
+The only disk device is actually the rootfs, as it corresponds to the <root>
+tag from the os definition
+There is one serial device of type `pty` with target port `0`, however the target port is
+not relevant depending on the version of firecracker.
+Below there are 2 net interfaces defined of type ethernet, each connected
+to already existing tapa devices
```
Comment (alindima) [inline docs/drvfc.rst]:
```suggestion
to already existing tap devices
```

File: docs/drvfc.rst [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+In the `<os>` tag we specified the source file for the kernel binary, the kernel command line.
+The only architecture supported for now is `x86_64` with the `hvm` type.
+The `<root>` tag specifies which of the following disk devices is used as the rootfs by matching the
+dev mount point attribute in the `<target>` tag.
+
+For the devices section:
+The only disk device is actually the rootfs, as it corresponds to the <root>
+tag from the os definition
+There is one serial device of type `pty` with target port `0`, however the target port is
+not relevant depending on the version of firecracker.
```
Comment (alindima) [inline docs/drvfc.rst]: what do you mean?

File: docs/drvfc.rst [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+         <target dev='vda' bus='virtio' />
+         <readonly/>
+      <disk>
+
+
+   ``<source>``
+      Mandatory tag that specifies the path to the backing file for the disk device
+
+   ``<target>``
+      Mandatory tag that.
```
Comment (alindima) [inline docs/drvfc.rst]: accidental sentence end

File: docs/drvfc.rst [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+      Specifies the drive id under the ``dev`` attribute.
+      Currently only virtio devices are supported, so this tag should be present: ``bus='virtio'``
```
Comment (alindima) [inline docs/drvfc.rst]: virtio over mmio*

File: docs/drvfc.rst [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+   ::
+
+      <serial type='pty'>
+         <target port='0'/>
+      </serial>
+
+
+   ``<target>``
+      Optional tag that specifies the port to be useed for the serial. However this property is currently ignored in firecracker `v0.*`
```
Comment (alindima) [inline docs/drvfc.rst]:
```suggestion
      Optional tag that specifies the port to be used for the serial. However this property is currently ignored in firecracker `v0.*`
```

Comment (alindima) [reply docs/drvfc.rst]: add details at the top about what Firecracker version this was tested on and is supported on

File: docs/drvfc.rst [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+   ``<cmdline>``
+      The command line arguments the vm will boot with
+
+   ``<root>``
+      The value of this tag tells the driver which one of the disk devices
+      is the rootfs one. It matches this value with the attribute `dev` on the ``<target>`` tag of
+      the device
+
+``<devices>``
```
Comment (alindima) [inline docs/drvfc.rst:113]: what about network interfaces?

File: docs/drvfc.rst [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+    </devices>
+  </domain>
+
+In the above example we specified the domain type to be `firecracker`,
+we specified the name of the domain, together with the memory and vcpus allocated.
+In the `<os>` tag we specified the source file for the kernel binary, the kernel command line.
+The only architecture supported for now is `x86_64` with the `hvm` type.
+The `<root>` tag specifies which of the following disk devices is used as the rootfs by matching the
+dev mount point attribute in the `<target>` tag.
+
```
Comment (alindima) [inline docs/drvfc.rst:74]: you may add a link to the docs about building a kernel and rootfs for FC: https://github.com/firecracker-microvm/firecracker/blob/main/docs/rootfs-and-kernel-setup.md

File: docs/meson.build (hunk 1 above)
Comment (felixp72) [reply docs/meson.build:115]: Yes, the html file is built from the rst file. The html file will be found in the `build` folder after compilation.

File: docs/drvfc.rst (hunk 6 above)
Comment (Rpd-Strike) [reply docs/drvfc.rst:80]: In this section I am explaining the example XML

File: docs/drvfc.rst (hunk 12 above)
Comment (felixp72) [reply docs/drvfc.rst:74]: Done

File: docs/drvfc.rst (hunk 11 above)
Comment (felixp72) [reply docs/drvfc.rst:113]: Added now.

File: docs/meson.build (hunk 1 above)
Comment (alindima) [reply docs/meson.build:115]: well yes but I see that other drivers commit their compiled html files in the repo. so we should include this in the PR before merging it

File: docs/drvfc.rst (hunk 3 above)
Comment (alindima) [reply docs/drvfc.rst]: there still are some sentences that don't end in a period

File: docs/drvfc.rst [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+Expected connection URI would be
+
+``fc:///system``
+
+
+Example guest domain XML configurations
+=======================================
+
+The following docs were written using Firecracker version ``v0.26``
```
Comment (alindima) [inline docs/drvfc.rst]: isn't it v0.25?

We should state the versions this driver works on (min v0.25.0 and lower than v1.0)

Comment (felixp72) [reply docs/drvfc.rst]: It works with both AFAIK, I will update this.

**Review by alindima — APPROVED** [review]

**Pull request merged**